jax/random.py

```diff
@@ -374,6 +374,9 @@ def _uniform(key, shape, dtype, minval, maxval) -> jnp.ndarray:
 
   minval = lax.convert_element_type(minval, dtype)
   maxval = lax.convert_element_type(maxval, dtype)
+  minval = jnp.broadcast_to(minval, shape)
+  maxval = jnp.broadcast_to(maxval, shape)
```

Comment:
To avoid unnecessary data duplication, I believe we could use
```
minval = lax.broadcast_to_rank(minval, len(shape))
```
``jnp.broadcast_to`` will instantiate a full array of the given shape, which would be wasteful in cases with a single `minval`/`maxval` generating a large array.

Reply:
Done